### ISSUE TITLE
Custom button for drag along with moveable controls

### ISSUE THREAD

**Issue by iprasadg** [daybrush/moveable#468]
## Environments
* Framework name: Angular
* Framework version: 8
* Moveable Component version: 0.21.5
* Testable Address(optional):

## Description
I want to add a custom button for drag along with moveable controls, and want to reader that custom button at the top of DOM so that it will be visible outside the container. How can I do this?  Here are the screenshots for better understanding

![Custom Button For Drag](https://user-images.githubusercontent.com/59184039/117572847-75cee680-b0f2-11eb-8f31-a874cda4a168.png)
![Hidden](https://user-images.githubusercontent.com/59184039/117572855-7a939a80-b0f2-11eb-89fc-76e21759bd2c.png)

**Comment by daybrush** [comment]
@iprasadg 

I am curious about how you made a custom button. Is it included in ables?

**Comment by iprasadg** [comment]
@daybrush No, I have written a separate code for it. I am using moveable from its early launch at that time ables are not there, so I have written my own code to use custom handle/buttons for drag. But now moveable supports custom buttons, still I am struggling to use ables in angular. Can you please help me?

**Comment by ryantando** [comment]
@daybrush Any idea how to implement this? custom able to drag the component

**Comment by iprasadg** [comment]
@daybrush can you please help me with this?

**Comment by daybrush** [comment]
@iprasadg @ryantando 


The component of moveable has a z-index of 2000.
Make the z-index equal to that button, or

I recommend making it custom able.
https://daybrush.com/moveable/storybook/?path=/story/make-custom-able--dimensionviewable

**Comment by iprasadg** [comment]
@daybrush 
If the container has overflow then z-index wont work. So yes custom able is the solution but how to do it in angular? The example shown is in react

**Comment by daybrush** [comment]
@iprasadg 

Custom Able in Vanilla.
https://codepen.io/daybrush/pen/JjXmbmb

It is a react code, but if that is converted, it becomes createElement. Use the second argument, createElement.

**Comment by ctrl-alt-deleted** [comment]
@daybrush  thanks for the example,  this cleared few things for me 👍🏼

**Comment by tvanrijn** [comment]
I'm trying to accomplish something similar, using a custom button for drag. But I only want to drag when actually using that button, not when clicking on the target component. Is that possible?

**Comment by ctrl-alt-deleted** [comment]
@tvanrijn   if u add custom button and then give it a class/id.    then u  can set this property on your moveable object:

`dragTarget = document.querySelector('.custom-button');`

**Comment by trananhanc99** [comment]
@ctrl-alt-deleted i did custom with property `ables` then I set property `dragTarget = SVGElement | HTMLElement` but it's not working. If u have any resolve. pls help me.

**Comment by ctrl-alt-deleted** [comment]
@trananhanc99  it should be similar  to this example:

`  const customAble = {
            name: "custom",
            props: {},
            events: {},
            render(moveable, React) {
                const rect = moveable.getRect();
                return React.createElement(
                    "button",
                    {
                        key: "able-key",
                        type: "button",
                        title: "Close",
                        onClick: () => {
                            console.log('click handler');
                        },
                    },
                    'btn text'
                );
            },
        };
        let movableProp = {
            ....
            ables: [customAble],
            props: {
                custom: true,
            },
        };`

**Comment by trananhanc99** [comment]
Thanks @ctrl-alt-deleted  but i want move element with button move
I have example on codesandbox: https://codesandbox.io/s/sad-johnson-hicfpd?file=/src/App.tsx
I don't know why it's not working.
if u have idea please help me. Thanks a lot

**Comment by trananhanc99** [comment]
@daybrush Thank you!

**Comment by ctrl-alt-deleted** [comment]
@trananhanc99  thanks for the example,  from my understanding this should work.....   although it been while when i did  this, also I used vanilla JS.

**Comment by trananhanc99** [comment]
@daybrush Thanks for your help. I will try it on
